hoomd/mpcd/test/rejection_sphere_geometry_filler_test.cc

```diff
@@ -0,0 +1,156 @@
+    std::cout << "Number of virtual Particles after filling one time: " << pdata->getNVirtual() << "\n";
+        {
+        ArrayHandle<Scalar4> h_pos(pdata->getPositions(), access_location::host, access_mode::read);
+        ArrayHandle<Scalar4> h_vel(pdata->getVelocities(), access_location::host, access_mode::read);
+        ArrayHandle<unsigned int> h_tag(pdata->getTags(), access_location::host, access_mode::read);
+
+        // ensure first particle did not get overwritten
+        CHECK_CLOSE(h_pos.data[0].x, 1, tol_small);
+        CHECK_CLOSE(h_pos.data[0].y, -2, tol_small);
+        CHECK_CLOSE(h_pos.data[0].z,  3, tol_small);
+        CHECK_CLOSE(h_vel.data[0].x, 123, tol_small);
+        CHECK_CLOSE(h_vel.data[0].y, 456, tol_small);
+        CHECK_CLOSE(h_vel.data[0].z, 789, tol_small);
+        UP_ASSERT_EQUAL(h_tag.data[0], 0);
+
+        // check if the particles have been placed outside the confinement
+        unsigned int N_out(0);
+        for (unsigned int i=pdata->getN(); i < pdata->getN() + pdata->getNVirtual(); ++i)
+            {
+            // tag should equal index on one rank with one filler
+            UP_ASSERT_EQUAL(h_tag.data[i], i);
+            // type should be set
+            UP_ASSERT_EQUAL(__scalar_as_int(h_pos.data[i].w), 1);
+
+            Scalar3 pos = make_scalar3(h_pos.data[i].x, h_pos.data[i].y, h_pos.data[i].z);
+            const Scalar r2 = dot(pos, pos);
+            if (r2 > Scalar(25))
+                ++N_out;
+            }
+        UP_ASSERT_EQUAL(N_out, pdata->getNVirtual());
+        }
+
+    /*
+     * Fill the volume again, which should approximately double the number of virtual particles
+     */
+    filler->fill(1);
+    std::cout << "Number of virtual Particles after filling twice: " << pdata->getNVirtual() << "\n";
+        {
+        ArrayHandle<Scalar4> h_pos(pdata->getPositions(), access_location::host, access_mode::read);
+        ArrayHandle<Scalar4> h_vel(pdata->getVelocities(), access_location::host, access_mode::read);
+        ArrayHandle<unsigned int> h_tag(pdata->getTags(), access_location::host, access_mode::read);
+
+        // check if the particles have been placed outside the confinement
+        unsigned int N_out(0);
+        for (unsigned int i=pdata->getN(); i < pdata->getN() + pdata->getNVirtual(); ++i)
+            {
+            // tag should equal index on one rank with one filler
+            UP_ASSERT_EQUAL(h_tag.data[i], i);
+            // type should be set
+            UP_ASSERT_EQUAL(__scalar_as_int(h_pos.data[i].w), 1);
+
+            Scalar3 pos = make_scalar3(h_pos.data[i].x, h_pos.data[i].y, h_pos.data[i].z);
+            const Scalar r2 = dot(pos, pos);
+            if (r2 > Scalar(25))
+                ++N_out;
+            }
+        UP_ASSERT_EQUAL(N_out, pdata->getNVirtual());
+        }
+
+    /*
+     * Test the average properties of the virtual particles.
+     */
+    // initialize variables for storing avg data
+    unsigned int N_avg(0);
```

Comment:
If you are going to be averaging, use a `Scalar` or you will end up doing integer math.